openbb_platform/providers/sec/openbb_sec/models/latest_financial_reports.py

```diff
@@ -0,0 +1,261 @@
+    )
+    report_type: Optional[str] = Field(
+        default=None,
+        description="Return only a specific form type. Default is all quarterly, annual, and current reports."
+        + f" Choices: {', '.join(report_type_choices)}.",
+    )
+
+    @field_validator("report_type", mode="before", check_fields=False)
+    @classmethod
+    def validate_report_type(cls, v):
+        """Validate the report type."""
+        if v is None:
+            return v
+        rpts = v.split(",")
+        for rpt in rpts:
+            if rpt not in report_type_choices:
+                raise ValueError(
+                    f"Invalid report type: {rpt}. Choices: {', '.join(report_type_choices)}"
+                )
+        return v
+
+
+class SecLatestFinancialReportsData(LatestFinancialReportsData):
+    """SEC Latest Financial Reports Data."""
+
+    items: Optional[str] = Field(
+        default=None, description="Item codes associated with the filing."
+    )
+    index_headers: str = Field(description="URL to the index headers file.")
+    complete_submission: str = Field(
+        description="URL to the complete submission text file."
+    )
+    metadata: Optional[str] = Field(
+        default=None, description="URL to the MetaLinks.json file, if available."
+    )
+    financial_report: Optional[str] = Field(
+        default=None, description="URL to the Financial_Report.xlsx file, if available."
+    )
+
+
+class SecLatestFinancialReportsFetcher(
+    Fetcher[SecLatestFinancialReportsQueryParams, list[SecLatestFinancialReportsData]]
+):
+    """SEC Latest Financial Reports Fetcher."""
+
+    @staticmethod
+    def transform_query(params: dict[str, Any]) -> SecLatestFinancialReportsQueryParams:
+        """Transform the query params."""
+        return SecLatestFinancialReportsQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: SecLatestFinancialReportsQueryParams,
+        credentials: Optional[dict[str, str]],
+        **kwargs: Any,
+    ) -> list[dict]:
+        """Extract the raw data from the SEC."""
+        # pylint: disable=import-outside-toplevel
+        from datetime import timedelta  # noqa
+        from openbb_core.provider.utils.helpers import amake_request
+        from warnings import warn
+
+        today = dateType.today()
+        query_date = query.date if query.date is not None else today
+
+        if query_date.weekday() > 4:
+            query_date -= timedelta(days=query_date.weekday() - 4)
+
+        date = query_date.strftime("%Y-%m-%d")
+
+        SEARCH_HEADERS = {
+            "User-Agent": "my real company name definitelynot@fakecompany.com",
+            "Accept-Encoding": "gzip, deflate",
+        }
+
+        forms = (
+            query.report_type
+            if query.report_type is not None
+            else (
+                "1-K%2C1-SA%2C1-U%2C1-Z%2C1-Z-W%2C10-D%2C10-K%2C10-KT%2C10-Q%2C10-QT%2C11-K%2C11-KT%2C15-12B%2C15-12G%2C"
```

Comment:
Can this be something we define in the helpers or the like if it will be reused?

Reply:
Unclear if it will be used elsewhere, it is what you get with this group selection:

<img width="688" alt="Screenshot 2024-11-07 at 7 14 08 PM" src="https://github.com/user-attachments/assets/0ec74a86-cef2-4341-aa7f-6d8f456c588c">

Reply:
Potentially this could be rounded down into the "discovery.filings" endpoint, but I'm not sure they fill the same purposes entirely.